domain/autocert/state/state_test.go

```diff
@@ -80,7 +80,7 @@ func (s *stateSuite) TestRetrieveNoCert(c *gc.C) {
 
 	// Retrieve an arbitrary non existent cert.
 	_, err := st.Get(context.Background(), "cert1")
-	c.Assert(err, gc.ErrorMatches, "autocert cert1: not found")
+	c.Assert(err, gc.ErrorMatches, "acme/autocert: certificate cache miss")
```

Comment:
I don't like asserting error strings as they are not meant for machine consumption. We should assert on type.

`c.Assert(err, jc.ErrorIs, errors.NotFound)`

Reply:
absolutely, done!